index.js

```diff
@@ -12,6 +12,7 @@ module.exports = (input, needle, replacement) => {
 	let ret = '';
 	let matchCount = 0;
 	let prevIndex = 0;
+	const startIndex = (typeof fromIndex === 'number' && fromIndex > 0) ? fromIndex : 0;
```

Comment:
Also, maybe have an early return here if the fromIndex is larger than the string.